internal/rbd/rbd_util.go

```diff
@@ -2071,3 +2071,28 @@ func strategicActionOnLogFile(ctx context.Context, logStrategy, logFile string)
 		log.ErrorLog(ctx, "unknown cephLogStrategy option %q: hint: 'remove'|'compress'|'preserve'", logStrategy)
 	}
 }
+
+// genVolFromVolIDWithMigration populate a rbdVol structure based on the volID format.
+func genVolFromVolIDWithMigration(
+	ctx context.Context, volID string, cr *util.Credentials, secrets map[string]string) (*rbdVolume, error) {
+	var err error
+	var rv *rbdVolume
+	if isMigrationVolID(volID) {
+		pmVolID, pErr := parseMigrationVolID(volID)
+		if pErr != nil {
+			return nil, pErr
+		}
+		rv, pErr = genVolFromMigVolID(ctx, pmVolID)
+		if pErr != nil {
+			return nil, pErr
+		}
+	} else {
+		rv, err = GenVolFromVolID(ctx, volID, cr, secrets)
+		rv.Destroy()
+		if err != nil {
+			return nil, err
+		}
```

Comment:
```suggestion
		if err != nil {
		rv.Destroy()
			return nil, err
		}
```

and from the caller in case of non error destroy need to be called